### PR TITLE
Update santizeForSerialization to normalize parameters for Python client

### PR DESCRIPTION
**Pull request by geekerzp**
* Refactor santizeForSerialization to static method.
* Sanitize header/query/form/body parameters.
* Sanitize datetime.datetime and datetime.date to iso8601 format string.
* Sanitize swagger model to dict.

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/python/swagger.mustache [hunk 1]
```diff
@@ -107,34 +107,34 @@ class ApiClient:
...
+    If obj is str, int, long, float, bool, return directly.
+    If obj is datetime.datetime, datetime.date convert to string in iso8601 format.
+    If obj is list, santize each element in the list.
+    If obj is dict, return the dict.
+    If obj is swagger model, return the properties dict.
+    """
+    if isinstance(obj, type(None)):
       return None
-    elif type(obj) in [str, int, long, float, bool]:
+    elif isinstance(obj, (str, int, long, float, bool, file)):
```
Comment (geekerzp) [inline modules/swagger-codegen/src/main/resources/python/swagger.mustache]: `isinstance` is better than `type`
ref: http://docs.quantifiedcode.com/python-code-patterns/readability/do_not_compare_types_use_isinstance.html?highlight=isinstance

File: modules/swagger-codegen/src/main/resources/python/swagger.mustache [hunk 2]
```diff
@@ -55,8 +55,8 @@ class ApiClient:
       # Need to remove None values, these should not be sent
       sentQueryParams = {}
       for param, value in queryParams.items():
-        if value != None:
```
Comment (geekerzp) [inline modules/swagger-codegen/src/main/resources/python/swagger.mustache]: PEP8: E711
comparison to None shoule be `if cond is not None`

**Comment by fehguy** [comment]
This looks good.

**Comment by geekerzp** [comment]
@fehguy hi Tony, this is another pr about python datetime parse of me, please have a look
https://github.com/swagger-api/swagger-codegen/pull/527
Thanks!

**Pull request merged**